include/boost/url/impl/url_view.ipp

```diff
@@ -818,4 +822,46 @@ operator<<(
 } // urls
 } // boost
 
+//----------------------------------------------------------
+//
+// std::hash specialization
+//
+//----------------------------------------------------------
+
+std::size_t
+std::hash<::boost::urls::url_view>::operator()(
+    ::boost::urls::url_view const& u) const noexcept
+{
+#if BOOST_URL_ARCH == 64
+    std::uint64_t const prime = 0x100000001B3ULL;
+    std::uint64_t hash  = 0xcbf29ce484222325ULL;
+#else
+    std::uint64_t const prime = 0x01000193UL;
+    std::uint64_t hash  = 0x811C9DC5UL;
+#endif
+    hash += salt_;
+    using parts = ::boost::urls::detail::parts_base;
+    ::boost::urls::detail::ci_digest(
+        u.get(parts::id_scheme), hash, prime);
+    ::boost::urls::detail::digest_encoded(
+        u.get(parts::id_user), hash, prime);
+    ::boost::urls::detail::digest_encoded(
+        u.get(parts::id_pass), hash, prime);
+    ::boost::urls::detail::digest_encoded(
+        u.get(parts::id_host), hash, prime);
+    ::boost::urls::detail::digest(
+        u.port(), hash, prime);
+    ::boost::urls::detail::normalized_path_digest(
```

Comment:
One thing here is the children are being digested from right to left. I don't know if this can cause any problem. The symmetry with `operator==` still holds.

I don't think there's a way to do that from left to right for the same reasons we discussed for `url_view::compare`

Reply:
dude we do not "digest children"